From 14eab9d5c0a15d082a154b9cdbe50a73c20fbf9e Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Thu, 17 Feb 2022 19:58:44 +0100
Subject: [PATCH 1/1] orjson fixes

---
 Cargo.lock | 4 ++--
 Cargo.toml | 5 ++++-
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 50d2487..91deac9 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -109,9 +109,9 @@ checksum = "1aab8fc367588b89dcee83ab0fd66b72b50b72fa1904d7095045ace2b0c81c35"
 
 [[package]]
 name = "libc"
-version = "0.2.117"
+version = "0.2.118"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "e74d72e0f9b65b5b4ca49a346af3976df0f9c61d550727f349ecd559f251a26c"
+checksum = "06e509672465a0504304aa87f9f176f2b2b716ed8fb105ebe5c02dc6dce96a94"
 
 [[package]]
 name = "libm"
diff --git a/Cargo.toml b/Cargo.toml
index 2a43490..adb4d89 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -71,7 +71,10 @@ encoding_rs = { version = "0.8", default_features = false }
 inlinable_string = { version = "0.1" }
 itoa = { version = "1", default_features = false }
 once_cell = { version = "1", default_features = false }
-pyo3 = { version = "0.15", default_features = false, features = ["extension-module"]}
+pyo3 = { version = "0.15", default-features = false, features = [
+    "extension-module",
+    "nightly"
+] }
 ryu = { version = "1", default_features = false }
 serde = { version = "1", default_features = false }
 serde_json = { version = "^1.0.68", default_features = false, features = ["std", "float_roundtrip"] }
-- 
2.35.1

