From 188fd6fb3748b995ca8fb692a3de3f133a22726b Mon Sep 17 00:00:00 2001
From: Jens Reidel <jens@troet.org>
Date: Wed, 28 Oct 2020 21:42:53 +0100
Subject: [PATCH] support bytes in send_str

---
 aiohttp/client_ws.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/aiohttp/client_ws.py b/aiohttp/client_ws.py
index 37d5ac73..fe5a5715 100644
--- a/aiohttp/client_ws.py
+++ b/aiohttp/client_ws.py
@@ -153,8 +153,8 @@ class ClientWebSocketResponse:
         await self._writer.pong(message)
 
     async def send_str(self, data: str, compress: Optional[int] = None) -> None:
-        if not isinstance(data, str):
-            raise TypeError("data argument must be str (%r)" % type(data))
+        if not isinstance(data, (str, bytes, bytearray)):
+            raise TypeError("data argument must be str or byte-ish (%r)" % type(data))
         await self._writer.send(data, binary=False, compress=compress)
 
     async def send_bytes(self, data: bytes, compress: Optional[int] = None) -> None:
-- 
2.26.2

