From 9e33738c28ddbb8042d66ca98908c66fe7e78ee3 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Sun, 26 Sep 2021 11:12:06 +0200
Subject: [PATCH 1/1] Support orjson

---
 sentry_sdk/utils.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sentry_sdk/utils.py b/sentry_sdk/utils.py
index 8fb03e0..7378295 100644
--- a/sentry_sdk/utils.py
+++ b/sentry_sdk/utils.py
@@ -1,5 +1,4 @@
 import base64
-import json
 import linecache
 import logging
 import os
@@ -10,6 +9,7 @@ import re
 
 from datetime import datetime
 
+import orjson
 import sentry_sdk
 from sentry_sdk._compat import urlparse, text_type, implements_str, PY2
 
@@ -47,7 +47,7 @@ BASE64_ALPHABET = re.compile(r"^[a-zA-Z0-9/+=]*$")
 def json_dumps(data):
     # type: (Any) -> bytes
     """Serialize data into a compact JSON representation encoded as UTF-8."""
-    return json.dumps(data, allow_nan=False, separators=(",", ":")).encode("utf-8")
+    return orjson.dumps(data)
 
 
 def _get_debug_hub():
-- 
2.31.1

