From f7f154843bbd7128c796e9d767f6550d1479dcc1 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Sun, 26 Sep 2021 11:12:06 +0200
Subject: [PATCH] Support orjson

---
 sentry_sdk/utils.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sentry_sdk/utils.py b/sentry_sdk/utils.py
index 4fd53e9..5d444b6 100644
--- a/sentry_sdk/utils.py
+++ b/sentry_sdk/utils.py
@@ -1,5 +1,4 @@
 import base64
-import json
 import linecache
 import logging
 import os
@@ -18,6 +17,7 @@ try:
 except ImportError:
     _PARTIALMETHOD_AVAILABLE = False
 
+import orjson
 import sentry_sdk
 from sentry_sdk._compat import PY2, PY33, PY37, implements_str, text_type, urlparse
 from sentry_sdk._types import MYPY
@@ -54,7 +54,7 @@ BASE64_ALPHABET = re.compile(r"^[a-zA-Z0-9/+=]*$")
 def json_dumps(data):
     # type: (Any) -> bytes
     """Serialize data into a compact JSON representation encoded as UTF-8."""
-    return json.dumps(data, allow_nan=False, separators=(",", ":")).encode("utf-8")
+    return orjson.dumps(data)
 
 
 def _get_debug_hub():
-- 
2.39.1

